src/main/java/org/apache/maven/buildcache/BuildCacheMojosExecutionStrategy.java

```diff
@@ -388,7 +388,13 @@ boolean isParamsMatched(
                 return false;
             }
 
-            if (!Strings.CS.equals(currentValue, expectedValue)) {
+            // Apply ignorePattern filtering to expected value if it's an array string representation
+            String filteredExpectedValue = expectedValue;
+            if (trackedProperty.getIgnorePattern() != null && expectedValue.startsWith("[") && expectedValue.endsWith("]")) {
+                filteredExpectedValue = filterArrayString(expectedValue, trackedProperty.getIgnorePattern());
```

Comment:
I would avoid implementing this logic in opposite direction. In the event of an infrequent pattern change, the cached value will be rebuilt and later reused. Dissecting the cache to ensure it can be accepted, seems not to be a big win, appears counterintuitive and also might adds possibility of additional, difficult to trace errors (when the cache is restored  incorrectly).